tiledb/sm/c_api/tiledb.cc

```diff
@@ -258,7 +259,9 @@ int32_t tiledb_array_schema_alloc(
 
   // Create a new ArraySchema object
   (*array_schema)->array_schema_ = make_shared<tiledb::sm::ArraySchema>(
-      HERE(), static_cast<tiledb::sm::ArrayType>(array_type));
+      HERE(),
+      make_shared<tiledb::sm::MemoryTracker>(HERE()),
```

Comment:
This is a place you'll want to instantiate a new `MemoryTracker::Type::ARRAY_CREATE` instance of MemoryTracker. It'd be something like:

```c++
auto memory_tacker = ctx->ctx_->create_memory_tracker();
memory_tracker->set_type(MemoryTracker::Type::ARRAY_CREATE);
```